src/core/mac/mac.cpp

```diff
@@ -1286,7 +1286,7 @@ void Mac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, otError aError
         break;
 
     case kOperationTransmitPoll:
-        assert(aFrame.GetAckRequest());
+        assert((aError != OT_ERROR_NONE) || aFrame.GetAckRequest());
```

Comment:
Do we also need to check for errors in the logic above because it also used `aFrame`?
```
    // Determine whether to re-transmit a broadcast frame.
    aFrame.GetDstAddr(dstAddr);	    aFrame.GetDstAddr(dstAddr);
    ...
```

Reply:
Yes. But that' a bit tricker. Will need to think and we can probably address it from another PR. 

On other errors like CCA failure, we do want to resend a broadcast frame (that's the point of the code to ensure we send broadcast more than once to make it more reliable).

The problem I see is that we may get an `ABORT` from two different paths. Radio platform may give us an `ABORT` (in this case a broadcast frame is valid and prepared and I think should be re-transmitted) or the frame may not be prepared in first place where `Mac` itself calls `HandleTransmitDone()` with `ABORT` (this case the frame is not prepared and is the previous/older frame and should not be re-transmitted).

Reply:
Closing this PR in favor of https://github.com/openthread/openthread/pull/4265.